packages/edit-post/src/components/header/plugin-more-menu-item/index.js

```diff
@@ -26,6 +26,37 @@ const PluginMoreMenuItem = ( { onClick = noop, ...props } ) => (
 	</PluginsMoreMenuGroup>
 );
 
+/**
+ * Renders a menu item in `Plugins` group in `More Menu` drop down, and can be used to as a button or link depending on the props provided.
+ * The text within the component appears as the menu item label.
+ *
+ * @param {Object} props Component properties.
+ * @param {string} [props.href] When `href` is provided then the menu item is represented as an anchor rather than button. It corresponds to the `href` attribute of the anchor.
+ * @param {string|Element} [props.icon=inherits from the plugin] The [Dashicon](https://developer.wordpress.org/resource/dashicons/) icon slug string, or an SVG WP element, to be rendered to the left of the menu item label.
+ * @param {Function} [props.onClick=noop] The callback function to be executed when the user clicks the menu item.
+ * @param {...*} [props.other] Any additional props are passed through to the underlying [MenuItem](/packages/components/src/menu-item/README.md) component.
```

Comment:
Wasn't sure how to document passed-through props. An alternative to this will be to mention it in the description. If folks understand this, I'm happy to go with this approach.